src/crimson/osd/osd_operations/client_request.cc

```diff
@@ -68,6 +68,39 @@ void ClientRequest::may_set_prev_op()
   }
 }
 
+ClientRequest::interruptible_future<> ClientRequest::send_incremental_map(PG& pg)
```

Comment:
I don't think any of this is necessary -- there's no ordering relationship at all between maps sent by the osd to catch up the client and IO requests.

Reply:
Thanks! This strips one of two two reasons for maintaining the order: the externally, client-visible order-of-responses.

However, there is still the second, internal one: the implicit contract between `ClientRequest` and `OpSequencer`. The latter expects `finish_op()` to be always called in the order.

Reply:
Sending the incremental map should really be independent of the client request response, they really aren't particularly related.

Reply:
the contract is fixed by #41882.

Reply:
Nah, I think it's about a different thing: handling PG-targeting `MOSDOp`.

Also, there are a few other early exits in the `ClientRequest`'s flow as well as the recovery-on-`EIO`.

Reply:
ahh, right! we also need to ensure that they behave in those cases.